### PR TITLE
Refactored code layout, removed hard requirement for discovery, added sample keyboard code

### PR DESCRIPTION
**Pull request by voodootikigod**
A lot of initial cleanup from the great first pass provided by @FluffyJack 
* Included a new sample that allows you to drive the drone with WASD keyboard commands
* Refactored directory layout to be more like other modules
* Cleaned up code base to adhere to JSHint
* Added a grunt file and JSHint so it can be linted
* Removed the need for providing a specific UUID by allowing for a findFirst functionality.

### CODE REVIEW

File: SamplesAndTools/index.js [hunk 1]
```diff
@@ -1,35 +1,70 @@
...
+        delay: 5000,
+        task: function () {
+          process.exit(0);
+        }
+      }
+    ]);
+
   });
-}
+});
```
Comment (rwaldron) [inline SamplesAndTools/index.js]: This looks so much nicer now!

File: README.md [hunk 2]
```diff
@@ -37,64 +58,116 @@ yourDrone.connect(function() {
 ```
 ### Taking off, moving, and landing
 
-We're now going to create a function that takes a drone and then by using `setTimeout` creates a timed sequence of calls to actions on the drone.
+We're now going to create a function that takes a drone and then by using a sequence of `temporal` tasks creates a timed sequence of calls to actions on the drone.
 
-It's important that you use `setTimeout` between calls to make sure you give it enough time to complete the previous action, sometimes it freaks it out if you don't wait.
+We recommend using `temporal` over a series of `setTimeout` chained calls for your sanity. Please abide by this when playing with the drone and ESPECIALLY if filing a ticket.
```
Comment (FluffyJack) [inline README.md]: Love it! And all the fixes to my quickly hacked together docs. :)

File: SamplesAndTools/index.js (hunk 1 above)
Comment (FluffyJack) [inline SamplesAndTools/index.js]: SOOOO MUCH NICER! The previous code is what happens when Ruby devs don't play with node for a long while.

**Comment by FluffyJack** [comment]
Love it, this was hacked together very quickly over a weekend, and I haven't done proper node for a fair while, so this is fantastic stuff, thanks very very much! Merging now.

**Pull request merged**